server/opts_test.go

```diff
@@ -2383,6 +2384,127 @@ func TestParsingLeafNodeRemotes(t *testing.T) {
 			t.Fatalf("Expected %v, got %v", expected, opts.LeafNode.Remotes[0])
 		}
 	})
+
+	t.Run("shuffle remote urls no effect", func(t *testing.T) {
```

Comment:
Unfortunately, you would have to change those tests if you move the shuffling. You would even have to add test with direct options (so 4 tests total): 2 with ProcessConfigFile() followed by NewServer() and then check remotes' urls array, and 2 with creation Options and setting options directly and then calling NewServer() and checking remotes.
Let me know if you need more details on that.